main.tf

```
@@ -2,8 +2,9 @@ provider "aws" {
 }
 
 locals {
-  iam_role_arn = var.iam_role_arn == null ? join("", aws_iam_role.this.*.arn) : var.iam_role_arn
-  record_all   = length(var.include_resource_types) == 0 ? true : false
+  iam_role_arn   = var.iam_role_arn == null ? join("", aws_iam_role.this.*.arn) : var.iam_role_arn
+  record_all     = length(var.include_resource_types) == 0 && length(var.exclude_resource_types) == 0 ? true : false
```

Don't actually need the ternary, since we're just setting the resulting bool...

```suggestion
  record_all     = length(var.include_resource_types) == 0 && length(var.exclude_resource_types) == 0
```